pkg/authentication/tokens.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package authentication
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/oauth2"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+)
+
+type JWKSTokenVerifier struct {
+	clientID string
```

Comment:
Just a **NONBLOCKING** question (probably I miss some knowledge context): why does the verifier have an attribute `clientID`? Did not see where it is used for.

Reply:
You are right, I put it there for consistency with the other verifier, and with the thought in mind that it could be helpful for next iterations. But I'll remove it, and if needed we'll add it back. Good catch!

Reply:
I just pushed the change.